### ISSUE TITLE
Precomputed transforms (ModalTransform, HelmholtzCache)

### ISSUE THREAD

**Issue by jejjohnson** [jejjohnson/somax#24]
## Summary

For multilayer models, expensive decompositions must be computed once and reused. Define the transform abstractions that store precomputed results as equinox modules.

Parent: Sub-epic: Core API Framework

## Design

```python
class ModalTransform(eqx.Module):
    Cl2m: Float[Array, "nl nl"]       # layer-to-mode
    Cm2l: Float[Array, "nl nl"]       # mode-to-layer
    eigenvalues: Float[Array, "nl"]   # modal eigenvalues
    rossby_radii: Float[Array, "nl"]  # Rossby deformation radii [m]

    @staticmethod
    def from_physics(H, g_prime, f0) -> "ModalTransform":
        A = _build_coupling_matrix(H, g_prime)
        lambd, R = jnp.linalg.eigh(A)
        L = jnp.linalg.inv(R)
        return ModalTransform(Cl2m=L, Cm2l=R, eigenvalues=lambd, ...)

    def to_modal(self, x): return jnp.einsum("lm,...mxy->...lxy", self.Cl2m, x)
    def to_layer(self, x): return jnp.einsum("lm,...mxy->...lxy", self.Cm2l, x)
```

## Generalization

The pattern generalizes to:
- **HelmholtzCache**: Precomputed spectral eigenvalues for elliptic solvers
- **Capacitance matrices**: For masked domain solvers
- **Sigma/isopycnal coordinate transforms**: Vertical coordinate mappings

All follow the same pattern: computed once in `__init__` from static config, stored on model. Small matrices (nl x nl), negligible storage.

## Package Location

`src/somax/_src/transforms/`

## Acceptance Criteria

- [ ] ModalTransform with `from_physics` factory
- [ ] `to_modal` / `to_layer` roundtrip is identity
- [ ] Rossby radii computed correctly for known configurations
- [ ] HelmholtzCache for precomputed spectral eigenvalues
- [ ] Tests verify eigendecomposition against known solutions
